kernel/limit.c

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// #define _GNU_SOURCE
+// #include <assert.h>
+// #include <sched.h>
+// #include <stdlib.h>
+// #include <string.h>
+// #include <sys/wait.h>
+#include <math.h>
+#include <sys/resource.h>
+
+#include <myst/eraise.h>
+#include <myst/fdtable.h>
+#include <myst/kernel.h>
+#include <myst/limit.h>
+#include <myst/mmanutils.h>
+#include <myst/process.h>
+
+int myst_limit_set_default(struct rlimit rlimits[])
+{
+    int ret = 0;
+
+    if (!rlimits)
+        ERAISE(-EFAULT);
+
+    // RLIMIT_CPU (match linux)
+    rlimits[RLIMIT_CPU].rlim_cur = RLIM_INFINITY;
+    rlimits[RLIMIT_CPU].rlim_max = RLIM_INFINITY;
+
+    // RLIMIT_FSIZE (match linux)
+    rlimits[RLIMIT_FSIZE].rlim_cur = RLIM_INFINITY;
+    rlimits[RLIMIT_FSIZE].rlim_max = RLIM_INFINITY;
+
+    // RLIMIT_DATA (70% of mman region)
```

Comment:
Curious if theres any rationale behind this number?

Reply:
This was the suggested limit from Bo.